### PR TITLE
Block export support

### PR DESCRIPTION
**Pull request by zhencliu**
ID: 1911380

### CODE REVIEW

**Comment by zhencliu** [comment]
qemu-nbd regression tests:
(1/7) type_specific.io-github-autotest-qemu.access_with_remote_storage.with_ncat.with_inet: PASS (468.08 s)
 (2/7) type_specific.io-github-autotest-qemu.access_with_remote_storage.with_ncat.with_socket: PASS (465.57 s)
 (3/7) type_specific.io-github-autotest-qemu.access_with_remote_storage.with_unix_socket: PASS (297.81 s)
 (4/7) type_specific.io-github-autotest-qemu.access_with_remote_storage.with_nmap: PASS (5.62 s)
 (5/7) type_specific.io-github-autotest-qemu.access_with_remote_storage.with_qemu_io.drop_connection: PASS (4.93 s)
 (6/7) type_specific.io-github-autotest-qemu.access_with_remote_storage.with_qemu_io.drop_sync: PASS (5.07 s)
 (7/7) type_specific.io-github-autotest-qemu.access_with_remote_storage.with_guestfish: PASS (122.32 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2021-01-20T08.50-e0c0b14/results.html
JOB TIME   : 1372.11 s

nbd internal server regression tests:
block-export-add/block-export-del (since qemu 5.2)
(1/4) type_specific.io-github-autotest-qemu.blockdev_inc_backup_pull_mode_vm_down: FAIL: qemu hangs (215.03 s)
 (2/4) type_specific.io-github-autotest-qemu.blockdev_inc_backup_pull_mode_diff: PASS (182.43 s)
 (3/4) type_specific.io-github-autotest-qemu.blockdev_inc_backup_pull_mode_vm_reboot: PASS (244.85 s)
 (4/4) type_specific.io-github-autotest-qemu.blockdev_inc_backup_pull_mode: PASS (188.52 s)
nbd-server-add/nbd-server-remove (before qemu 5.2)
 (1/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_pull_mode_diff.q35: PASS (193.60 s)
 (2/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_pull_mode.q35: PASS (191.22 s)
 (3/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_pull_mode_vm_down.q35: FAIL: qemu hangs (296.54 s)
 (4/4) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_pull_mode_vm_reboot.q35: PASS (348.92 s)

**Comment by zhencliu** [comment]
hi @YongxueHong @ngu-niny , can you please help to review this patch? Thanks.
@ngu-niny , can you please perform regression testing on ppc? you may pick up several cases from the regression tests above. Thank you.

File: virttest/nbd.py [hunk 1]
```diff
@@ -106,8 +119,12 @@ def export_image(qemu_nbd, filename, local_image, params):
...
             )
 
-    if params.get('nbd_export_bitmap'):
-        cmd_dict['bitmap'] = '-B %s' % params['nbd_export_bitmap']
+    # multiple bitmaps can be exported
+    bitmap_setting = ''
+    for bm in params.objects('nbd_export_bitmap'):
+        bitmap_setting += ' -B %s' % bm
+    if bitmap_setting:
+        cmd_dict['bitmap'] = bitmap_setting.strip()
```
Comment (YongxueHong) [inline virttest/nbd.py:127]:
```suggestion
    if params.get('nbd_export_bitmaps'):
        cmd_dict['bitmap'] = "".join([" -B %s" % _ for _ in params['nbd_export_bitmap'].split()])
```
Your codes look good for me, just I prefer using `nbd_export_bitmaps` to clarify your intention to 
tell the user who could assign multiple bitmaps by the name of `nbd_export_bitmaps`, like images, vms, nics, etc.
Thanks.

Comment (zhencliu) [reply virttest/nbd.py:127]: You are right, yongxue, at the very beginning, I still use the original param name, which can avoid some case-level code change, in the end, I forgot to make the change, I will update it later. Thanks much.

**Comment by YongxueHong** [comment]
BTW, your commit message, for each line, does not need to add an extra blank, just at the top.
Thanks.

**Comment by zhencliu** [comment]
hi @YongxueHong , all changes including the commit msg have been done, can you please help to make a double check, thanks.
 (1/7) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_pull_mode_diff.q35: PASS (175.40 s)
 (2/7) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_xpt_allocation_depth.with_qemu_nbd.q35: PASS (91.73 s)
 (3/7) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_xpt_allocation_depth.with_block_export.q35: PASS (88.63 s)
 (4/7) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_pull_mode.q35: PASS (164.55 s)
 (5/7) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_xpt_bitmap.q35: PASS (85.73 s)
 (6/7) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_xpt_multiple_bitmaps.with_qemu_nbd.q35: PASS (66.09 s)
 (7/7) Host_RHEL.m8.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_xpt_multiple_bitmaps.with_block_export.q35: PASS (65.60 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2021-02-07T04.03-78cb9dd/results.html
JOB TIME   : 741.68 s

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**